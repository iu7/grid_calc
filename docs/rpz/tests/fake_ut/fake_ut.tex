% !TeX document-id = {fb8a2ef5-cdaf-49da-b79d-0a8152e677cd}
% !TeX TS-program = XeLaTeX
\documentclass[a4paper,12pt]{report}

% polyglossia should go first!
\usepackage{polyglossia} % multi-language support
\setmainlanguage{russian}
\setotherlanguage{english}

\usepackage{amsmath} % math symbols, new environments and stuff
\usepackage{unicode-math} % for changing math font and unicode symbols
\usepackage[style=english]{csquotes} % fancy quoting
\usepackage{microtype} % for better font rendering
\usepackage{hyperref} % for refs and URLs
\usepackage{graphicx} % for images (and title page)
\usepackage{geometry} % for margins in title page
\usepackage{tabu} % for tabulars (and title page)
\usepackage[section]{placeins} % for float barriers
\usepackage{titlesec} % for section break hooks
\usepackage{listings} % for listings 
\usepackage{upquote} % for good-looking quotes in source code (used for custom languages)
\usepackage{xcolor} % colors!
\usepackage{enumitem} % for unboxed description labels (long ones)
\usepackage{caption}
\usepackage{multirow}
\usepackage{varwidth}
\usepackage{longtable}

\defaultfontfeatures{Mapping=tex-text} % for converting "--" and "---"
\setmainfont{CMU Serif}
\setsansfont{CMU Sans Serif}
\setmonofont{CMU Typewriter Text}
\setmathfont{XITS Math}
\MakeOuterQuote{"} % enable auto-quotation

% new page and barrier after section, also phantom section after clearpage for
% hyperref to get right page.
% clearpage also outputs all active floats:
\newcommand{\sectionbreak}{\phantomsection}
\newcommand{\subsectionbreak}{\FloatBarrier}
\renewcommand{\thesection}{\arabic{section}} % no chapters
\numberwithin{equation}{section}
%\usetikzlibrary{shapes,arrows,trees}

\newcommand{\itemtt}[1][]{\item[\texttt{#1}:]} % tt-ed items (for protocol descriptions)

\definecolor{bluekeywords}{rgb}{0.13,0.13,1}
\definecolor{greencomments}{rgb}{0,0.5,0}
\definecolor{turqusnumbers}{rgb}{0.17,0.57,0.69}
\definecolor{redstrings}{rgb}{0.5,0,0}
\setmonofont{Consolas} %to be used with XeLaTeX or LuaLaTeX
\definecolor{bluekeywords}{rgb}{0,0,1}
\definecolor{greencomments}{rgb}{0,0.5,0}
\definecolor{redstrings}{rgb}{0.64,0.08,0.08}
\definecolor{xmlcomments}{rgb}{0.5,0.5,0.5}
\definecolor{types}{rgb}{0.17,0.57,0.68}

\lstloadlanguages{bash, python, Java}

\lstset{
  frame=none,
  xleftmargin=2pt,
  stepnumber=1,
  numbers=left,
  numbersep=5pt,
  numberstyle=\ttfamily\tiny\color[gray]{0.3},
  belowcaptionskip=\bigskipamount,
  captionpos=b,
  escapeinside={*'}{'*},
  language=python,
  tabsize=2,
  emphstyle={\bf},
  commentstyle=\it,
  stringstyle=\mdseries\rmfamily,
  showspaces=false,
  keywordstyle=\bfseries\rmfamily,
  columns=flexible,
  basicstyle=\small\sffamily,
  showstringspaces=false,
  morecomment=[l]\%,
  breaklines=true,
  showlines=true
}
\renewcommand\lstlistingname{Листинг}

\date{\today}

\makeatletter
\let\thetitle\@title
\let\theauthor\@author
\let\thedate\@date
\makeatother

\makeatletter
\AtBeginDocument{%
    \expandafter\renewcommand\expandafter\subsection\expandafter{%
        \expandafter\@fb@secFB\subsection
    }%
}
\makeatother

\begin{document}
\section{Модульное тестирование}
Статический класс DataMethods:\\
Предствляет собой интерфейс бекенда данных.
\begin{table}[h!]
\caption{Методы класса DataMethodsFilter}
\begin{tabu} to \textwidth {|c|X|}
\hline
Название метода & Примечания \\
\multirow{3}{*}{get\_item} & param: table [ str ] \\
                           & param: value\_json[ dict ] \\
                           & Возвращает все элементы таблицы с именем table, удовлетворяющие фильтру value\_json \\
\hline
\multirow{3}{*}{put\_item} & param: table [ str ] \\
                           & param: value\_json [ dict ] \\
                           & Изменяет все элементы таблицы с именем table, удовлетворяющие фильтру value\_json \\
\hline
\multirow{3}{*}{delete\_item} & param: table [ str ] \\
                           & param: value\_json [ dict ] \\
                           & Удаляет все элементы таблицы с именем table, удовлетворяющие фильтру value\_json \\
\hline
\end{tabu}
\end{table}

\begin{table}[h]
\caption{Категория 1 -- Тестирование метода получения данных}
\begin{tabu} to \textwidth {|c|X|}
\hline
Название теста & TestGetItem \\ \hline
Тестируемый метод & get\_item \\ \hline
Описание теста & Проверка получения данных из БД, пустой таблицы trait \\ \hline
Ожидаемый результат & Cловарь с пустым списком в ключе ``result'' \\ \hline
Степень важности & Фатальная \\ \hline
Результат теста & Тест пройден \\ \hline
\end{tabu}
\end{table}

\clearpage

\begin{table}[h]
\caption{Категория 2 -- Тестирование метода изменения данных}
\begin{tabu} to \textwidth {|c|X|}
\hline
Название теста & TestPutItem \\ \hline
Тестируемый метод & put\_item \\ \hline
\multirow{4}{*}{Описание теста} & Проверка изменения данных в БД, в таблице trait, предварительно наполненной записями \\
                                & name = ``test\_name'', version = ``1.0'' \\
                                & name = ``test\_name'', version = ``2.0'', \\
                                & по фильтру \{``name'':``test\_name'',``changes'': \{``version'':``3.0''\}\} \\
\hline
Ожидаемый результат & Словарь \{``count'': 2\}, отражающий наличие двух произведенных изменений в БД \\ \hline
Степень важности & Фатальная \\ \hline
Результат теста & Тест пройден \\ \hline
\end{tabu}
\end{table}

\begin{table}[h]
\caption{Категория 3 -- Тестирование метода удаления данных}
\begin{tabu} to \textwidth {|c|X|}
\hline
Название теста & TestDeleteItem \\ \hline
Тестируемый метод & delete\_item \\ \hline
\multirow{4}{*}{Описание теста} & Проверка удаления данных в БД, в таблице trait, предварительно наполненной записями \\
                                & name = ``test\_name'', version = ``1.0'' \\
                                & name = ``test\_name'', version = ``2.0'', \\
                                & по фильтру \{``name'':``test\_name''\} \\
\hline
Ожидаемый результат & Словарь \{``count'': 2\}, отражающий наличие двух произведенных удалений в БД \\ \hline
Степень важности & Фатальная \\ \hline
Результат теста & Тест пройден \\ \hline
\end{tabu}
\end{table}


Вывод по результатам тестирования:\\
Все тесты пройдены успешно, класс готов к использованию.

\end{document}