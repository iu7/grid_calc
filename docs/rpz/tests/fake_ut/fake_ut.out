\BOOKMARK [1][-]{section*.1}{Модульное тестирование}{}% 1
